src/main/java/com/google/research/bleth/servlets/WriteHardCodedBoardStatesServlet.java

```diff
@@ -0,0 +1,73 @@
+package com.google.research.bleth.servlets;
+
+import com.google.research.bleth.services.DatabaseService;
+import com.google.research.bleth.simulator.Board;
+import com.google.research.bleth.simulator.DummyBeacon;
+import com.google.research.bleth.simulator.Location;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+/**
+ * Servlet for testing the board visualization.
+ *
+ * Clear all real and estimated board state entities (corresponding to the demo simulation id) from datastore.
+ *
+ * Generate 10 random real boards and 10 random estimated boards in a demo 5*5 simulation,
+ * by locating dummy beacons in random locations.
+ *
+ * Write generated board state to datastore.
+ *
+ */
+@WebServlet("/write-hard-coded-board-state")
+public class WriteHardCodedBoardStatesServlet extends HttpServlet {
+
+    private DatabaseService datastore = DatabaseService.getInstance();
+    private Board realBoard;
+    private Board estimatedBoard;
+    private DummyBeacon firstDummyBeacon;
+    private DummyBeacon secondDummyBeacon;
+    private DummyBeacon thirdDummyBeacon;
+
+    @Override
+    public void init() {
+        realBoard = new Board(5, 5);
+        estimatedBoard = new Board(5, 5);
+        firstDummyBeacon = new DummyBeacon(0, new Location((int) (Math.random() * 3),(int) (Math.random() * 3)));
+        secondDummyBeacon = new DummyBeacon(1, new Location((int) (Math.random() * 3),(int) (Math.random() * 3)));
+        thirdDummyBeacon = new DummyBeacon(2, new Location((int) (Math.random() * 3),(int) (Math.random() * 3)));
+    }
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+        // Clear datastore.
+        String simulationId = "demo-tracing-sim-1";
+        datastore.deleteAllSimulationRealBoardStates(simulationId);
+        datastore.deleteAllSimulationEstimatedBoardStates(simulationId);
+
+        for (int round = 0; round < 10; round++) {
+            // Clear boards.
+            realBoard = new Board(5, 5);
+            estimatedBoard = new Board(5, 5);
+
+            // Randomly locate dummy beacons on both boards.
+            realBoard.placeAgent(new Location((int) (Math.random() * 3),(int) (Math.random() * 3)), firstDummyBeacon);
```

Comment:
Ouch. I don't like: you just placed the beacon in a random location on the real board, that's different from the beacon's internal location. So - we created an inconsistent state.

Please use the `move` methods so the beacon's internal state reflects the real board.

Reply:
Oh my.. exactly the kind of mistakes I'm having on exams :(